### ISSUE TITLE
`no-trailing-spaces` removes blank lines from KDoc when the line isn't preceded by `*`

### ISSUE THREAD

**Issue by RBusarow** [pinterest/ktlint#1376]
## Expected Behavior
Blank lines inside a KDoc should not be removed (at least not by a rule which handles trailing spaces).

## Observed Behavior

Note that asterisks on each line aren't technically required, even though they're convention.

```kotlin
/**
 Paragraph 1 which should be followed by a blank line.

 Paragraph 2 which should have a blank line before it.
 */
class MyClass
```
Ktlint wants to change it to this:
```kotlin
/**
 Paragraph 1 which should be followed by a blank line.
 Paragraph 2 which should have a blank line before it.
 */
class MyClass
```

## Your Environment

* Version of ktlint used: `0.44.0`
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Gradle plugin
